gdpr/impl.go

```diff
@@ -210,8 +210,8 @@ func (p *permissionsImpl) checkPurpose(consent tcf2.ConsentMetadata, vendor api.
 		return true
 	}
 
-	purposeAllowed := consent.PurposeAllowed(purpose) && (weakVendorEnforcement || (vendor.Purpose(purpose) && consent.VendorConsent(vendorID)))
-	legitInterest := consent.PurposeLITransparency(purpose) && (weakVendorEnforcement || (vendor.LegitimateInterest(purpose) && consent.VendorLegitInterest(vendorID)))
+	purposeAllowed := p.consentEstablished(consent, vendor, vendorID, purpose, weakVendorEnforcement)
+	legitInterest := p.legitInterestEstablished(consent, vendor, vendorID, purpose, weakVendorEnforcement)
```

Comment:
I'm on the fence; this is a very good point. On the flip side, I like having the logic separate for consent established and legit interest established because it helps us verify we have sufficient test coverage (we want to make sure we exercise each of the conditions for both consent and legit interest). I’d like to punt on this decision and reevaluate your suggestion in a future PR that involves overhauling impl.go and impl_test.go . I’ll note it in that refactor story so I don’t forget.